src/Symfony/Component/Notifier/Bridge/Clickatell/ClickatellTransport.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Clickatell;
+
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Kevin Auvinet <k.auvinet@gmail.com>
+ */
+final class ClickatellTransport extends AbstractTransport
+{
+    protected const HOST = 'api.clickatell.com';
+
+    private $authToken;
+    private $from;
+
+    public function __construct(string $authToken, ?string $from = null, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->authToken = $authToken;
+        $this->from = $from;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $endpoint = sprintf('https://%s/rest/message', $this->getEndpoint());
+
+        $response = $this->client->request('POST', $endpoint, [
+            'headers' => [
+                'Accept' => 'application/json',
+                'Authorization' => 'Bearer '.$this->authToken,
+                'Content-Type' => 'application/json',
+                'X-Version' => 1,
+            ],
+            'json' => [
+                'from' => $this->from ?? '',
+                'to' => [$message->getPhone()],
+                'text' => $message->getSubject(),
+            ],
+        ]);
+
+        if (202 === $response->getStatusCode()) {
+            return new SentMessage($message, (string) $this);
+        }
+
+        $content = $response->toArray(false);
```

Comment:
```suggestion
        $content = $response->toArray(false);

        if (202 === $response->getStatusCode()) {
            $sentMessage = new SentMessage($message, (string) $this);
            $sentMessage->setMessageId((string) ANY IDEA?)
            
            return $sentMessage;
        }

```

Did you get some id back in the response which can be set?

Reply:
Yes, done !